workflow/controller/controller.go

```diff
@@ -598,9 +629,24 @@ func (wfc *WorkflowController) signalContainers(namespace string, podName string
 		_ = signal.SignalContainer(wfc.restConfig, pod, c.Name, sig)
 	}
 	if pod.Spec.TerminationGracePeriodSeconds == nil {
-		return 30 * time.Second, nil
+		return 30 * time.Second
+	}
+	return time.Duration(*pod.Spec.TerminationGracePeriodSeconds) * time.Second
+}
+
+func (wfc *WorkflowController) getPod(namespace string, podName string) (*apiv1.Pod, error) {
```

Comment:
Moved the `getPod` function for clarity since `processNextPodCleanupItem` no longer uses the `getPod` function.